### PR TITLE
Extra content filters

### PR DESCRIPTION
**Pull request by danigm**
This PR is based on: https://github.com/endlessm/kolibri/pull/10

https://phabricator.endlessm.com/T32509

### CODE REVIEW

File: kolibri/plugins/learn/assets/src/views/ChannelRenderer/CustomContentRenderer.vue
Comment (manuq) [inline kolibri/plugins/learn/assets/src/views/ChannelRenderer/CustomContentRenderer.vue]: For better API design I think this should be a separate method. I imagine 2 methods:
- getChannelMetadata -- title, description, thumbnail
- getChannelFilterOptions -- availableAuthors, availableTags, availableActivityTypes

Then if a custom presentation doesn't need filters, the second query is not done. What do you think?

File: kolibri/plugins/learn/assets/src/views/ChannelRenderer/CustomContentRenderer.vue
Comment (manuq) [inline kolibri/plugins/learn/assets/src/views/ChannelRenderer/CustomContentRenderer.vue]: There is no need to join here. The serializer takes care of it.

Comment (manuq) [reply kolibri/plugins/learn/assets/src/views/ChannelRenderer/CustomContentRenderer.vue]: Also I don't understand very well when the activity type (like Watch) gets converted to the kind (like Video), or if the names are used interchangable. There is a map here https://github.com/learningequality/kolibri/blob/6903afec0ca3ebdf65fbf0eb96ef5348ed9fc2d2/kolibri/core/content/api.py#L245-L250

File: packages/hashi/src/kolibri.js
Comment (manuq) [inline packages/hashi/src/kolibri.js]: Since it's an array and not a single activity type, the name should be in plural: `options.activityTypes`.

**Review by manuq — CHANGES_REQUESTED** [review]
Looks very promising. Let me know what do you think about having a separate API just for the filters available options.

File: kolibri/plugins/learn/assets/src/views/ChannelRenderer/CustomContentRenderer.vue
Comment (danigm) [reply kolibri/plugins/learn/assets/src/views/ChannelRenderer/CustomContentRenderer.vue]: I didn't know about the serializers, in that case I've updated the PR to remove the three `join` that I used there.

File: kolibri/core/assets/src/api-resources/channel.js
Comment (manuq) [inline kolibri/core/assets/src/api-resources/channel.js:14]: I don't understand why this is needed here.

**Review by manuq — CHANGES_REQUESTED** [review]
It seems you forgot to commit this in the vue file:
```
      this.hashi.on(events.CHANNELFILTEROPTIONSREQUESTED, message => {
        this.sendChannelFilterOptions.call(this, message);
      });
```

File: kolibri/core/assets/src/api-resources/channel.js
Comment (danigm) [reply kolibri/core/assets/src/api-resources/channel.js:14]: This is just a helper method that's to be able to do call `ChannelResource.fetchFilterOptions(this.topic.channel_id)` we can always do this call directly, but I saw that this is a pattern used in the [ContentResource](https://github.com/endlessm/kolibri/blob/release-v0.14.x/kolibri/core/assets/src/api-resources/contentNode.js#L8), so I implemented in the same way.

Comment (manuq) [reply kolibri/core/assets/src/api-resources/channel.js:14]: Ah excellent. It was suspicious because it's the first one in the channel resource.

**Review by manuq — APPROVED** [review]
Fantastic work! I just commited lint fixups for files changed after running pre-commit hook. Let's wait for the CI to run before merging.

**Comment by danigm** [comment]
It looks like there are some tests failing. I'll update the tests to fix all jobs in the CI.

**Pull request merged**
